File: isu-service/src/main/java/ru/itmo/soa/isu/IsuService.java
package ru.itmo.soa.isu;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import ru.itmo.soa.model.ApiModels.ActionResult;
import ru.itmo.soa.model.ApiModels.FormOfEducation;
import ru.itmo.soa.model.ApiModels.StudyGroup;
import ru.itmo.soa.model.ApiModels.StudyGroupRequest;

@ApplicationScoped
public class IsuService {
    @Inject
    StudyGroupsClient client;

    public ActionResult expelAll(int groupId) {
        client.delete(groupId);
        return new ActionResult("ALL_STUDENTS_EXPELLED",
                "Все студенты отчислены; пустая группа удалена", groupId);
    }

    public StudyGroup changeEducationForm(int groupId, FormOfEducation newForm) {
        StudyGroup current = client.get(groupId);
        StudyGroupRequest request = current.toRequest();
        request.setFormOfEducation(newForm);
        return client.update(groupId, request);
    }
}

File: src/main/kotlin/ru/itmo/soa/isu/IsuService.kt
package ru.itmo.soa.isu

import org.springframework.stereotype.Service
import ru.itmo.soa.api.dto.StudyGroupResponse
import ru.itmo.soa.domain.FormOfEducation

@Service
class IsuService(private val client: StudyGroupsClient) {
    fun expelAll(groupId: Int) = client.delete(groupId)

    fun changeEducationForm(groupId: Int, newForm: FormOfEducation): StudyGroupResponse {
        val current = client.get(groupId)
        return client.update(groupId, current.toRequest(newForm))
    }
}
